wagtail/signal_handlers.py

```diff
@@ -83,6 +83,9 @@ def update_reference_index_on_save(instance, **kwargs):
             break
 
         instance = getattr(instance, parental_keys[0].name)
+        if instance is None:
```

Comment:
@laymonage I can't quite see what failure case you have in mind here. As far as I can see, the only situation where a default value on `getattr` would come into play is if the `parental_keys[0].name` attribute is missing from `instance`... but `parental_keys` here comes from the instance's own `instance._meta.get_fields()`, so if a field name is defined in there it must surely exist in `instance`'s attributes, right?

Reply:
Ah, sorry, I think I meant handling a `DoesNotExist` exception that's thrown by Django if the foreign key refers to a nonexisting instance. Not quite sure if that's something possible in this context, though!

Reply:
Ah, OK. I do recall some situation where following a "dangling" relation resulted in a `DoesNotExist` exception rather than returning None as you might expect, but can't remember exactly when that happens. (Generic foreign keys, perhaps?)

I don't think that's something we need to consider here unless we get specific reports of it in the wild, though. Usually ParentalKeys will be defined with on_delete=CASCADE or PROTECT - a nullable ParentalKey is already a bit of a questionable edge case, and a ParentalKey without referential integrity is an edge case on top of an edge case :-)

Reply:
I see, that sounds reasonable!